### PR TITLE
Implement rain logic

### PR DESCRIPTION
**Pull request by olliewalsh**
Adds 3 modes for rain:
0 - Ignore it
1 - Dock when rain detected. If in automatic mode it will resume again when battery is full etc...
2 - Dock until dry. If in automatic mode will resume again when rain is no longer detected for OM_RAIN_DELAY_MINUTES
3 - Pause Automatic Mode. Pauses automatic mode when rain detected. Needs to be manually resumed

### CODE REVIEW

**Comment by Apehaenger** [comment]
Quite cool flexible implementation!
Will test it asap...

File: src/mower_logic/src/mower_logic/mower_logic.cpp [hunk 1]
```diff
@@ -534,6 +538,29 @@ void checkSafety(const ros::TimerEvent &timer_event) {
     dockingNeeded = true;
   }
 
+  // Rain detected is initialized to true and flips to false if rain is not detected
+  // continuously for rain_check_seconds. This is to avoid false positives due to noise
```
[the remark this answers is not in the document]
Comment (rovo89) [reply src/mower_logic/src/mower_logic/mower_logic.cpp:542]: A positive side of having this in higher level is that rain could be signalled externally without having to reimplement everything. But then again, I guess we could just set manual_pause_mowing...

[the remark this answers is not in the document]
Comment (olliewalsh) [reply src/mower_logic/src/mower_logic/mower_logic.cpp:542]: I just meant move to logic to deal with noise to the low level firmware so we get a stable true/false from the low level status

**Comment by Apehaenger** [comment]
Sorry to dirt your branch, but had problems creating an image for testing.

Finally could test.
Tested only the default mode "Ignore", "Dock" and "Dock until dry", but code is pretty clear.
All worked as expected.

Thanks a lot for this quite cool PR, I do know that a lot of people are waiting for this functionality!

**Pull request merged**